### PR TITLE
log file with a different instance root directory

### LINKED ISSUE #10372: log file with a different instance root directory
**What steps does it take to reproduce the issue?**
If you have a different instance root, you cannot execute the commands as you describe. (https://guides.dataverse.org/en/5.14/admin/metadataexport.html#batch-exports-through-the-api)
Because the log directory has been permanently programmed.
For example here:
https://github.com/IQSS/dataverse/blob/ec1b174dbc89edd4e40c9c61760523e800497754/src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java#L702
(I have already made an improvement and I will upload it right away.)

* When does this issue occur?
curl http://localhost:8080/api/admin/metadata/exportAll
curl http://localhost:8080/api/admin/metadata/reExportAll
curl http://localhost:8080/api/admin/metadata/clearExportTimestamps

* Which page(s) does it occurs on?

* What happens?
It is not executed at all and an error occurs.

* To whom does it occur (all users, curators, superusers)?

* What did you expect to happen?

**Which version of Dataverse are you using?**
It is the same source code in 5.13 to 6

**Any related open or closed issues to this bug report?**

**Screenshots:**

No matter the issue, screenshots are always welcome.

To add a screenshot, please use one of the following formats and/or methods described here:

* https://help.github.com/en/articles/file-attachments-on-issues-and-pull-requests
*

### PR DESCRIPTION
**Pull request by BenediktMeierUIT**
**What this PR does / why we need it**:
log file with a different instance root directory
**Which issue(s) this PR closes**:
Closes #10372 

**Special notes for your reviewer**:

**Suggestions on how to test this**:

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

**Is there a release notes update needed for this change?**:

**Additional documentation**:

### CODE REVIEW

**Comment by pdurbin** [comment]
@BenediktMeierUIT thanks for the pull request. Out of curiosity, what is your instance root?

**Comment by Louis-wr** [comment]
> @BenediktMeierUIT thanks for the pull request. Out of curiosity, what is your instance root?

Benedikt as been working with us on dataverse.no

**Comment by pdurbin** [comment]
Cool. I'm still wondering what you use as an instance root, but no rush. 😄 Oh! Maybe you're using Docker? 🤔 

We recently make a similar change at 8efb964 and as discussed [in Slack](https://iqss.slack.com/archives/C010LA04BCG/p1690921171172649?thread_ts=1690912135.247139&cid=C010LA04BCG), we know there are more places to fix, as you've done. Thanks!

**Comment by BenediktMeierUIT** [comment]
Yes, this is a Docker.
And I solved the errors straight away. 
https://github.com/IQSS/dataverse/commit/8efb96487d181f83726ac5a338a13692b1710850

I'm not quite sure about another point. Why it was done differently.
https://github.com/IQSS/dataverse/blob/ec1b174dbc89edd4e40c9c61760523e800497754/src/main/java/edu/harvard/iq/dataverse/DatasetVersionServiceBean.java#L896

https://github.com/IQSS/dataverse/blob/ec1b174dbc89edd4e40c9c61760523e800497754/src/main/java/edu/harvard/iq/dataverse/batch/util/LoggingUtil.java#L70

(I don't have access to Slack :-( )

**Comment by pdurbin** [comment]
Sorry, can you please spell out a little more how it's done differently? From a quick look it seems more or less the same to me.

**Comment by BenediktMeierUIT** [comment]
Sorry... 
It's the same, but it's inconsistent.
My question was/is, what is the guideline and how do you want it?

**Comment by BenediktMeierUIT** [comment]
At that point, it's done differently again:
https://github.com/IQSS/dataverse/blob/5caf436c6321a8a26749d3dbf8e9ea4dfcdde5f7/src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvesterServiceBean.java#L91
I was going to change the line like this:
`public static final String DATAVERSE_HARVEST_STOP_FILE="stopharvest_";`
and in the place:
https://github.com/IQSS/dataverse/blob/5caf436c6321a8a26749d3dbf8e9ea4dfcdde5f7/src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvesterServiceBean.java#L402
`String stopFileName = System.getProperty("com.sun.aas.instanceRoot") + File.separator + "logs" + File.separator + DATAVERSE_HARVEST_STOP_FILE + harvestingClient.getName() + "." + pid; `

**Comment by poikilotherm** [comment]
This should be a JVM option defaulting to the system property.

**Comment by pdurbin** [comment]
It's funny, just today I was in DatasetServiceBean.java debugging something and I copied the fix from this PR because I'm in Docker and was hitting a problem without it.

I'd say we definitely want this fix.

And yes, a JVM option/MicroProfile Config setting eventually. In this PR or later.

**Comment by Louis-wr** [comment]
Yes, A JVM option/MicroProfile Config  would be nice in the future but for now we should avoid overreaching this pull request and  keep it confine to the bug fix that it is.
In the near future once this is fixed it will be easy to replace all the `System.getProperty("com.sun.aas.instanceRoot") `with the appropriate newly created JVM option/MicroProfile Config.

**Comment by pdurbin** [comment]
Yes, sounds good to defer adding a JVM/MPConfig option until later.

**Comment by poikilotherm** [comment]
Adding the option is almost a one liner. Do you want me to create a small PR to this PR with it? @pdurbin if you leave a review after, I can do QA and we ship this one fast.

**Comment by BenediktMeierUIT** [comment]
If you give me the JVM/MPConfig name, I can change the format to what you want.

**Comment by pdurbin** [comment]
Sure, I'm happy to do a review and let @poikilotherm do the QA.

File: src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java [hunk 1]
```diff
@@ -699,7 +699,7 @@ public void exportAllDatasets(boolean forceReExport) {
         Integer countError = 0;
         String logTimestamp = logFormatter.format(new Date());
         Logger exportLogger = Logger.getLogger("edu.harvard.iq.dataverse.harvest.client.DatasetServiceBean." + "ExportAll" + logTimestamp);
-        String logFileName = "../logs" + File.separator + "export_" + logTimestamp + ".log";
+        String logFileName = System.getProperty("com.sun.aas.instanceRoot") + File.separator + "logs" + File.separator + "export_" + logTimestamp + ".log";
```
Comment (pdurbin) [inline src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java:702]: I can vouch for the need for this change. Without it, I couldn't get some code I was playing with to execute. I was exporting some JSON for troubleshooting: https://github.com/pdurbin/dataverse/commit/8adb28d67522936b7e3c6f0aa98e8ab720a1982a

Comment (poikilotherm) [reply src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java:702]: Sounds like within a container, the application has a different working directory than in a classic deployment. Which means `../logs` may point to some place that does not exist / the app user has no permissions to write to.

**Review by poikilotherm — CHANGES_REQUESTED** [review]
With the explanation via DM and in https://github.com/IQSS/dataverse/pull/10373/files#r1535665937, I understood better why this change is necessary. 

As we should be thorough with this change, can you fix the missing occurrences @BenediktMeierUIT ?

Those would be in `dataverse.authorization.AuthFilter` and `dataverse.harvest.client.HarvesterServiceBean`.

**Comment by BenediktMeierUIT** [comment]
That's right... 
I have changed it.

**Review by poikilotherm — APPROVED** [review]
Looks good to me now, thanks for the change! @pdurbin stage is yours :smile:

**Comment by pdurbin** [comment]
Tests are passing as as I mentioned above these changes are very similar to changes I made in https://github.com/pdurbin/dataverse/commit/8adb28d67522936b7e3c6f0aa98e8ab720a1982a

Merging.

**Pull request merged**